stl/inc/ranges

```diff
@@ -7656,6 +7662,401 @@ namespace ranges {
+
+            _NODISCARD_FRIEND constexpr _Iterator operator+(const _Iterator& _Lhs, const difference_type _Rhs) noexcept(
+                noexcept(_Iterator{*_Lhs._Parent, _Lhs._Inner + _Rhs})) // strengthened
+                requires random_access_range<_Base_t>
+            {
+                return _Iterator{*_Lhs._Parent, _Lhs._Inner + _Rhs};
+            }
+
+            _NODISCARD_FRIEND constexpr _Iterator operator+(const difference_type _Lhs, const _Iterator& _Rhs) noexcept(
+                noexcept(_Iterator{*_Rhs._Parent, _Rhs._Inner + _Lhs})) // strengthened
+                requires random_access_range<_Base_t>
+            {
+                return _Iterator{*_Rhs._Parent, _Rhs._Inner + _Lhs};
+            }
+
+            _NODISCARD_FRIEND constexpr _Iterator operator-(const _Iterator& _Lhs, const difference_type _Rhs) noexcept(
+                noexcept(_Iterator{*_Lhs._Parent, _Lhs._Inner - _Rhs})) // strengthened
+                requires random_access_range<_Base_t>
+            {
+                return _Iterator{*_Lhs._Parent, _Lhs._Inner - _Rhs};
+            }
+
+            // clang-format off
+            _NODISCARD_FRIEND constexpr difference_type operator-(const _Iterator& _Lhs, const _Iterator& _Rhs)
+                noexcept(noexcept(_Lhs._Inner - _Rhs._Inner)) // strengthened
+                requires sized_sentinel_for<_Ziperator<_IsConst>, _Ziperator<_IsConst>>
+            {
+                // clang-format on
+                return _Lhs._Inner - _Rhs._Inner;
+            }
+
+        private:
+            _NODISCARD constexpr auto _Dereference_closure() const noexcept {
+                return [this](const auto&... _Itrs) noexcept(noexcept(_STD invoke(*_Parent->_Function,
+                           *_Itrs...))) -> decltype(auto) { return _STD invoke(*_Parent->_Function, *_Itrs...); };
+            }
+
+            _NODISCARD constexpr auto _Random_access_closure(const difference_type _Where) const noexcept {
+                return [this, _Where]<class... _Iterator_types>(const _Iterator_types&... _Iters) noexcept(
+                           noexcept(_STD invoke(*_Parent->_Function,
+                               _Iters[static_cast<iter_difference_t<_Iterator_types>>(_Where)]...))) -> decltype(auto) {
+                    return _STD invoke(
+                        *_Parent->_Function, _Iters[static_cast<iter_difference_t<_Iterator_types>>(_Where)]...);
+                };
+            }
+        };
+
+        template <bool _IsConst>
+        class _Sentinel {
+        private:
+            friend zip_transform_view;
+
+            // So, here's an interesting MSVC bug: If we have _Sentinel::operator-() return
+            // range_difference_t<_Maybe_const<_IteratorConst, _Inner_view>> directly as suggested
+            // in the C++ standard, ADL does not seem to find the function in the test suite, and the
+            // program fails to compile. This does not seem to happen with Clang.

```

Comment:
This should be reported as an MSVC bug, and then this code should be marked as `TRANSITION`, citing the bug number. (We conventionally avoid literally saying "bug" in the product code, with only a few exceptions.)

Reply:
Alright, I've submitted a bug report containing a small reproduction case of the bug, removed that specific comment, and labeled all instances of `_Iterator_sentinel_difference_type` as `TRANSITION, DevCom-10253131`. However, I'm not sure if 10253131 is the correct bug number, since it doesn't seem like the Developer Community portal exposes these values publicly. I got that value by examining the `id` field of the JSON payload returned by the `https://sendvsfeedback2.azurewebsites.net/api/detailsV3/rootPost` endpoint.

Can you confirm that bug number 10253131 matches the issue at https://developercommunity.visualstudio.com/t/C23---Using-Type-Alias-in-Nested-Class/10253131? (The same number does appear in the URL, but I don't know if that's the bug number or some other identifier used by the feedback system.)

Reply:
Thanks for reporting it!

Yep, that's the right number. DevCom behaves in an interesting way - when you file a bug, it crafts a URL based on part of the given title (hence the `C23---Using-Type-Alias-in-Nested-Class` component), followed by the bug number. However, it doesn't actually care about that component, and will happily look up just `https://developercommunity.visualstudio.com/t/<number>`. This is how we get DevCom-10253131 to work with our [custom autolinks as mentioned in the wiki](https://github.com/microsoft/STL/wiki/Custom-Autolinks). If the autolink works, then the citation is correct.

Every DevCom bug *does* have an associated internal "FeedbackTicket" (with a different number) and most of those have an associated internal VSO bug (another different number) which are not publicly derivable given just the DevCom bug, but citing just the DevCom bug number in code and issues is totally cool.